tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -0,0 +1,346 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+# 
+# Summary: A test to pass SR-IOV Ethernet VFs to guest via libvirt. Both KVM & XEN hosts are supported.
+# Maintainer: Julie CAO <JCao@suse.com>
+ 
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+use utils;
+use virt_utils;
+use xen;
+use virt_autotest::virtual_network_utils qw(save_guest_ip test_network_interface);
+ 
+sub run_test {
+    my $self = shift;
+
+    #set up ssh, packages and iommu on host
+    prepare_host();
+ 
+    #get the SR-IOV device BDF and interface
+    my @host_pfs;
+    @host_pfs = get_sriov_ethernet_devices();
+    record_info("Find SR-IOV devices", "@host_pfs");
```

Comment:
Please use "Find SR-IOV network devices" or similar. And you do not have error handling here, if get_sriov_ethernet_devices returns empty the test should die immediately instead of continue to run.

Reply:
Good catch, add return 1. change to use find_xxx.

Reply:
Please ensure you have the desired test result after returning, pass, fail or softfailure.